src/namer.h

```diff
@@ -0,0 +1,129 @@
+#ifndef FLATBUFFERS_NAMER
+#define FLATBUFFERS_NAMER
+
+#include "flatbuffers/idl.h"
+#include "flatbuffers/util.h"
+
+namespace flatbuffers {
+
+// `Namer` applies style configuration to symbols in generated code. It manages
+// casing, escapes keywords, and object API naming.
+// TODO: Refactor all code generators to use this.
+class Namer {
+ public:
+  struct Config {
+    // Symbols in code.
+    Case types;
+    Case constants;
+    Case methods;
+    Case functions;
+    Case fields;
+    Case variants;
+
+    // Namespaces
+    Case namespaces;
+    std::string namespace_seperator;
+
+    // Object API.
+    std::string object_prefix;
+    std::string object_suffix;
+
+    // Keywords.
+    std::string keyword_prefix;
+    std::string keyword_suffix;
+
+    // Files.
+    Case filenames;
+    Case directories;
+    std::string output_path;
+    std::string filename_suffix;
+    std::string filename_extension;
+
+    // DO NOT SUBMIT
+    // comment/doc comment.
+  };
+  Namer(Config config, std::set<std::string> keywords)
+      : config_(config), keywords_(std::move(keywords)) {}
+
+  // Returns a mutable reference to the config so it can be modified **before**
+  // code generation. The intent is for this class to be const while generating
+  // code.
+  Config &GetConfig() { return config_; }
+
+  std::string Type(const std::string &s) const {
+    return Format(s, config_.types);
+  }
+  std::string Method(const std::string &s) const {
+    return Format(s, config_.methods);
+  }
+  std::string Constant(const std::string &s) const {
+    return Format(s, config_.constants);
+  }
+  std::string Function(const std::string &s) const {
+    return Format(s, config_.functions);
+  }
+  std::string Field(const std::string &s) const {
+    return Format(s, config_.fields);
+  }
+  std::string Variant(const std::string &s) const {
+    return Format(s, config_.variants);
+  }
+  std::string ObjectType(const std::string &s) const {
+    return config_.object_prefix + Type(s) + config_.object_suffix;
+  }
+  std::string Namespace(const std::string &s) const {
+    return Format(s, config_.namespaces);
+  }
+  // DO NOT SUBMIT: Add GetRelativeNamespaceTraversal?
+
+  std::string EscapeKeyword(const std::string &name) const {
```

Comment:
Not directly to this, but there have been cases where user-defined names clash with local variables in the runtime libs (e.g. #7133). It might be nice to have a Namer function that can escape those as well?

Reply:
I could just make it a generic "escape" method... thoughts?